core-ui/src/state/navigation/filters/permissions.js

```diff
@@ -0,0 +1,55 @@
+import pluralize from 'pluralize';
+
+export const doesResourceExist = ({
+  resourceGroupAndVersion,
+  resourceKind,
+  resourceIdList,
+}) => {
+  const resourceNamePlural = pluralize(resourceKind);
+
+  const resourcePathId = `/${
+    resourceGroupAndVersion === 'v1' ? 'api' : 'apis'
+  }/${resourceGroupAndVersion}/${resourceNamePlural}`.toLowerCase();
+
+  const doesExist = !!resourceIdList.includes(resourcePathId);
+
+  return doesExist;
+};
+
+export const doesUserHavePermission = (
+  permissions = ['get', 'list'],
```

Comment:
I'd go with only "list" for default - only this verb is required for deciding if we want to show or hide the navigation node. The same in `core/src/luigi-config/navigation/permissions.js`:51.